crates/sui-core/src/consensus_handler.rs

```diff
@@ -63,85 +64,94 @@ impl ConsensusHandler {
 impl ExecutionState for ConsensusHandler {
     /// This function will be called by Narwhal, after Narwhal sequenced this certificate.
     #[instrument(level = "trace", skip_all)]
-    async fn handle_consensus_transaction(
+    async fn handle_consensus_output(
         &self,
         // TODO [2533]: use this once integrating Narwhal reconfiguration
-        consensus_output: &Arc<ConsensusOutput>,
-        consensus_index: ExecutionIndices,
-        serialized_transaction: Vec<u8>,
+        consensus_output: ConsensusOutput,
     ) {
-        let index = Self::update_hash(
-            &self.last_seen,
-            consensus_index.clone(),
-            &serialized_transaction,
-        );
-        let index = if let Some(index) = index {
-            index
-        } else {
-            debug!(
+        let _scope = monitored_scope("HandleConsensusOutputFull");
+        let mut sequenced_transactions = Vec::new();
+        let mut seq = 0;
```

Comment:
For consistency with how it was implemented previously, perhaps you should move this `seq` inside the loop below, so that it will be index in the certificate

Reply:
Now the sequence is assigned for every transaction in the sub_dag, rather than for every transaction in the cert. This sequence generation is going to move to a separate function in the next PR.